examples/bridge-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,452 @@
+
+        emberAfReportingAttributeChangeCallback(dev->GetEndpointId(), ZCL_FIXED_LABEL_CLUSTER_ID, ZCL_LABEL_LIST_ATTRIBUTE_ID,
+                                                CLUSTER_MASK_SERVER, 0, ZCL_ARRAY_ATTRIBUTE_TYPE, buffer);
+    }
+}
+
+extern "C" void app_main()
+{
+    int err = 0;
+    // Initialize the ESP NVS layer.
+    err = nvs_flash_init();
+    if (err != CHIP_NO_ERROR)
+    {
+        ESP_LOGE(TAG, "nvs_flash_init() failed: %s", ErrorStr(err));
+        return;
+    }
+
+    CHIP_ERROR chip_err = CHIP_NO_ERROR;
+
+    // bridge will have own database named gDevices.
+    // Clear database
+    memset(gDevices, 0, sizeof(gDevices));
+
+    // Bridged devices 4
+    Device Light1("Light 1", "Office");
+    Device Light2("Light 2", "Office");
+    Device Light3("Light 3", "Kitchen");
+    Device Light4("Light 4", "Den");
+
+    // Whenever bridged device changes its state
+    Light1.SetChangeCallback(&HandleDeviceStatusChanged);
+    Light2.SetChangeCallback(&HandleDeviceStatusChanged);
+    Light3.SetChangeCallback(&HandleDeviceStatusChanged);
+    Light4.SetChangeCallback(&HandleDeviceStatusChanged);
+
+    Light1.SetReachable(true);
+    Light2.SetReachable(true);
+    Light3.SetReachable(true);
+    Light4.SetReachable(true);
+
+    CHIPDeviceManager & deviceMgr = CHIPDeviceManager::GetInstance();
+
+    err = deviceMgr.Init(&AppCallback);
+    if (err != CHIP_NO_ERROR)
+    {
+        ESP_LOGE(TAG, "device.Init() failed: %s", ErrorStr(err));
+        return;
+    }
+
+    InitServer();
+
+    // Set starting endpoint id where dynamic endpoints will be assigned, which
+    // will be the next consecutive endpoint id after the last fixed endpoint.
+    gFirstDynamicEndpointId = static_cast<chip::EndpointId>(
+        static_cast<int>(emberAfEndpointFromIndex(static_cast<uint16_t>(emberAfFixedEndpointCount() - 1))) + 1);
+    gCurrentEndpointId = gFirstDynamicEndpointId;
+
+    // Disable last fixed endpoint, which is used as a placeholder for all of the
+    // supported clusters so that ZAP will generated the requisite code.
+    emberAfEndpointEnableDisable(emberAfEndpointFromIndex(static_cast<uint16_t>(emberAfFixedEndpointCount() - 1)), false);
+
+    // Add lights 1..3 --> will be mapped to ZCL endpoints 2, 3, 4
+    AddDeviceEndpoint(&Light1, &bridgedLightEndpoint, DEVICE_TYPE_LO_ON_OFF_LIGHT);
+    AddDeviceEndpoint(&Light2, &bridgedLightEndpoint, DEVICE_TYPE_LO_ON_OFF_LIGHT);
+    AddDeviceEndpoint(&Light3, &bridgedLightEndpoint, DEVICE_TYPE_LO_ON_OFF_LIGHT);
+
+    // Remove Light 2 -- Lights 1 & 3 will remain mapped to endpoints 2 & 4
+    RemoveDeviceEndpoint(&Light2);
+
+    // Add Light 4 -- > will be mapped to ZCL endpoint 5
+    AddDeviceEndpoint(&Light4, &bridgedLightEndpoint, DEVICE_TYPE_LO_ON_OFF_LIGHT);
+
+    // Re-add Light 2 -- > will be mapped to ZCL endpoint 6
+    AddDeviceEndpoint(&Light2, &bridgedLightEndpoint, DEVICE_TYPE_LO_ON_OFF_LIGHT);
```

Comment:
If majority of the code here is exactly same for the Linux bridge example, would it make sense to move it to some common location, may be `examples/bridge-app/bridge-common/`. This is just an observation and I am not sure about this. Can you file an issue so that it can be discussed?

Reply:
I think this should be example-specific. ESP32 was written, with reference to Linux, hence they are the same, this use-case may or may not be the same for other platforms. However I have created the issue. #8458

Reply:
Yes, that's right, some part of it would be example specific. But, still APIs like `AddDeviceEndpoint`, `RemoveDeviceEndpoint`, different callbacks and utility functions can be made common.